packages/contracts-bedrock/src/safe/DeputyPauseModule.sol

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Safe
+import { GnosisSafe as Safe } from "safe-contracts/GnosisSafe.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+
+// Contracts
+import { EIP712 } from "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+
+// Libraries
+import { ECDSA } from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+
+// Interfaces
+import { ISemver } from "interfaces/universal/ISemver.sol";
+import { IDeputyGuardianModule } from "interfaces/safe/IDeputyGuardianModule.sol";
+import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
+
+/// @title DeputyPauseModule
+/// @notice Safe Module designed to be installed in the Foundation Safe which allows a specific
+///         deputy address to act as the Foundation Safe for the sake of triggering the
+///         Superchain-wide pause functionality. Significantly simplifies the process of triggering
+///         a Superchain-wide pause without changing the existing security model.
+contract DeputyPauseModule is ISemver, EIP712 {
+    /// @notice Error message for invalid deputy.
+    error DeputyPauseModule_InvalidDeputy();
+
+    /// @notice Error message for unauthorized calls.
+    error DeputyPauseModule_Unauthorized();
+
+    /// @notice Error message for nonce reuse.
+    error DeputyPauseModule_NonceAlreadyUsed();
+
+    /// @notice Error message for failed transaction execution.
+    error DeputyPauseModule_ExecutionFailed(string);
+
+    /// @notice Error message for the SuperchainConfig not being paused.
+    error DeputyPauseModule_SuperchainNotPaused();
+
+    /// @notice Error message for the call not being from the Foundation Safe.
+    error DeputyPauseModule_NotFromSafe();
+
+    /// @notice Struct for the Pause action.
+    /// @custom:field nonce Signature nonce.
+    struct PauseMessage {
+        bytes32 nonce;
+    }
+
+    /// @notice Struct for the DeputyAuth action.
+    /// @custom:field deputy Address of the deputy account.
+    struct DeputyAuthMessage {
+        address deputy;
+    }
```

Comment:
This needs to be in a struct for reasons related to EIP-712 I assume?

Reply:
Correct